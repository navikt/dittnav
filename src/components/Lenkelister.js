import React from 'react';
import { shape, arrayOf, string } from 'prop-types';
import { Normaltekst } from 'nav-frontend-typografi';
import { listOfActions, logLenkeTrykkAmplitude } from '../utils/amplitudeUtils';

const Lenkelister = (props) => (
  <div className="flere-tjenester">
    <nav className="flere-tjenester__links">
      {props.links.map(l => (
        <div className="flere-tjenester__link-container" key={l.url}>
          <Normaltekst>
            <a
              href={l.url}
              className="lenke flere-tjenester__link"
              onClick={() => logLenkeTrykkAmplitude(l.tittel, listOfActions.TrykkPaaLenkeliste)}
            >
              {l.tittel}
            </a>
          </Normaltekst>
        </div>
      ))}
    </nav>
  </div>
);

Lenkelister.propTypes = {
  links: arrayOf(shape({
    url: string.isRequired,
    tittel: string.isRequired,
  })),
};

Lenkelister.defaultProps = {
  links: [],
};

export default Lenkelister;
