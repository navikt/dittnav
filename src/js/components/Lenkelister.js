import React from 'react';
import { shape, arrayOf, string } from 'prop-types';
import { Normaltekst } from 'nav-frontend-typografi';
import { GoogleAnalyticsAction, GoogleAnalyticsCategory, trackEvent } from '../utils/GoogleAnalytics';

const Lenkelister = (props) => (
  <div className="flere-tjenester">
    <nav className="flere-tjenester__links">
      {props.links.map(l => (
        <div className="flere-tjenester__link-container" key={l.url}>
          <Normaltekst>
            <a
              href={l.url}
              data-ga="Dittnav/Lenkeliste"
              className="lenke flere-tjenester__link"
              onClick={() => trackEvent(
                GoogleAnalyticsCategory.Forside,
                `${GoogleAnalyticsAction.FlereTjenester}/${l.tittel}`,
                l.url,
              )}
            >
              {l.tittel}
            </a>
          </Normaltekst>
        </div>
      ))}
    </nav>
  </div>
);

Lenkelister.propTypes = {
  links: arrayOf(shape({
    url: string.isRequired,
    tittel: string.isRequired,
  })),
};

Lenkelister.defaultProps = {
  links: [],
};

export default Lenkelister;
