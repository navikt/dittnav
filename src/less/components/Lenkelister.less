.flere-tjenester {
  margin: 1.5rem;
  column-count: 1;

  &__subheader {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    margin-bottom: 1.25rem;
    text-align: center;

    &::before,
    &::after {
      content: '';
      flex: 1;
      height: 1px;
      background-color: lightgray;
    }

    span {
      flex-shrink: 0;
      margin: 0 1rem;
    }
  }

  &__link-container {
    margin-bottom: 1.5rem;
    text-align: center;
  }

  &__link {
    color: #0067c5;
    text-decoration: none !important;
    border-bottom: 1px solid lightgray;

    &:hover {
      border-bottom-color: @navBla;
    }

    .typo-normal:hover {
      border-bottom: 0;
    }

    .typo-normal:hover > p {
      border-bottom: 0;
    }
  }

  &__link-icon {
    margin-left: 0.5rem;
    position: relative;
  }

  p {
    text-align: center;
    display: inline-block;
    border-bottom: 1px solid transparent;
  }

  a {
    display: inline-block;
  }

  @media (min-width: @mobile-breakpoint) {
    column-count: 2;
  }

  @media (min-width: @tablet-breakpoint) {
    column-count: 3;
  }
}
