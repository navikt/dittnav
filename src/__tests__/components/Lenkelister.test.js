import * as React from 'react';
import Lenkelister from 'js/components/Lenkelister';
const ReactTestRenderer = require('react-test-renderer');

test('basic green Lenkelister test', () => {
  const component = ReactTestRenderer.create((<Lenkelister links={[]} />));
  expect(component.toJSON()).toMatchSnapshot();
});

test('render Lenkelister without links', () => {
  const component = ReactTestRenderer.create((<Lenkelister />));
  expect(component.toJSON()).toMatchSnapshot();
});

test('render a couple of links', () => {
  const links = [
    { url: '/mininnboks', tittel: 'Min innboks' },
    { url: '/saksoversikt', tittel: 'Innsyn og oversikt' },
  ];
  const component = ReactTestRenderer.create((<Lenkelister links={links} />));
  expect(component.toJSON()).toMatchSnapshot();
});
