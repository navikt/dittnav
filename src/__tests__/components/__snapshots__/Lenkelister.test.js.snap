// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic green Lenkelister test 1`] = `
<div
  className="flere-tjenester"
>
  <nav
    className="flere-tjenester__links"
  />
</div>
`;

exports[`render Lenkelister without links 1`] = `
<div
  className="flere-tjenester"
>
  <nav
    className="flere-tjenester__links"
  />
</div>
`;

exports[`render a couple of links 1`] = `
<div
  className="flere-tjenester"
>
  <nav
    className="flere-tjenester__links"
  >
    <div
      className="flere-tjenester__link-container"
    >
      <p
        className="typo-normal"
      >
        <a
          className="lenke flere-tjenester__link"
          data-ga="Dittnav/Lenkeliste"
          href="/mininnboks"
          onClick={[Function]}
        >
          Min innboks
        </a>
      </p>
    </div>
    <div
      className="flere-tjenester__link-container"
    >
      <p
        className="typo-normal"
      >
        <a
          className="lenke flere-tjenester__link"
          data-ga="Dittnav/Lenkeliste"
          href="/saksoversikt"
          onClick={[Function]}
        >
          Innsyn og oversikt
        </a>
      </p>
    </div>
  </nav>
</div>
`;
